buzz_el/graph_loader/knowledge_graph.py

```diff
@@ -0,0 +1,265 @@
+
+        Returns
+        -------
+        Set[str]
+            The annotation properties attribute.
+        """
+        return self._annotation_properties
+
+    @annotation_properties.setter
+    def annotation_properties(self, props: Set[str]) -> None:
+        """Setter for the annotation properties attribute.
+
+        Makes sure the annotation properties are ready to be used in a 
+        SPARQL query.
+
+        Parameters
+        ----------
+        props : Set[str]
+            The annotation properties.    
+        """
+        if props is not None:
+            annotation_properties = set()
+            # we need to add <uri> if a full URI is provided
+            for prop in props:
+                if is_valid_url(prop):
+                    annotation_properties.add(f"<{prop}>")
+                else:
+                    annotation_properties.add(prop)
+
+            self._annotation_properties = annotation_properties
+
+    def load_kg_from_file(self) -> None:
+        """Load the knowledge graph from the specified file."""
+        self.kg.parse(self._kg_file_path)
+
+    def sparql_endpoint(self, sparql_query: str) -> Iterable:
+        """SPARQL endpoint to query the knowledge graph.
+
+        Parameters
+        ----------
+        sparql_query : str
+            The SPARQL query to run.
+
+        Returns
+        -------
+        Iterable
+            Iterable over the query results.
+        """
+        res_iter = self.kg.query(sparql_query)
+        return res_iter
+
+    def build_patterns(self) -> List[Dict[str, str]]:
+        """Build the entity patterns based on the interpretation type.
+
+        Returns
+        -------
+        List[Dict[str, str]]
+            The entity patterns.
+
+        Raises
+        ------
+        NotImplementedError
+            Error for the interpretation types not yet implemented.
+        """
+        patterns = []
+
+        if self.interpretation_type == "RDFS":
+            raise NotImplementedError
+        elif self.interpretation_type == "OWL":
+            raise NotImplementedError
+        else: # default to RDF (even if wrong value for interpretation type)
+            patterns = self.build_patterns_from_rdf()
+
+        return patterns
+
+    def build_patterns_from_rdf(self) -> List[Dict[str, str]]:
+        """Build the entity patterns specifically for the interpretation type "RDF".
+
+        Returns
+        -------
+        List[Dict[str, str]]
+            The entity patterns.
+        """
+        query = self._build_ent_str_sparql_query()
+
+        sparql_res = self.sparql_endpoint(sparql_query=query)
+
+        patterns = []
+        for res in sparql_res:
+            patterns.append({
+                "label": "KG_ENT", 
+                "pattern": str(res["label"]),
+                "id": str(res["ent_uri"])
+            })
+
+        return patterns
+
+    def get_context(self, entity_uri: str) -> List[str]:
```

Comment:
How can we customize that in the future ? 
I am not a big fan of the implementation. 
We will have different  `_build_ent_context_sparql_query` function ? We have to add a callable as parameter ?

Reply:
That is a great question. I had the same, but I wanted to implement a first version. It should definitely change in the future.